code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+% version, collecting settings for enabled channels and streams
+% (stored in TD_SettingsOut, Power_SettingsOut, FFT_SettingsOut)
+%
+% Input: Folder path to Device* folder containing json files
+% Output: TD_SettingsOut, Power_SettingsOut, FFT_SettingsOut, metaData
+%
+% Requires convertTDcodes.m
+%%
+% Load in DeviceSettings.json file
+DeviceSettings = jsondecode(fixMalformedJson(fileread([folderPath filesep 'DeviceSettings.json']),'DeviceSettings'));
+%%
+% Fix format - Sometimes device settings is a struct or cell array
+if isstruct(DeviceSettings)
+    DeviceSettings = {DeviceSettings};
+end
+%%
+% UTC offset to determine timezone conversion
+metaData.UTCoffset = DeviceSettings{1,1}.UtcOffset;
+
+% Battery information, as of beginning of the current recording
+metaData.batteryLevelPercent = DeviceSettings{1}.BatteryStatus.batteryLevelPercent;
+metaData.batteryVoltage = DeviceSettings{1}.BatteryStatus.batteryVoltage;
+metaData.estimatedCapacity = DeviceSettings{1}.BatteryStatus.estimatedCapacity;
+metaData.batterySOC = DeviceSettings{1}.BatteryStatus.batterySOC;
+
+%%
+TD_SettingsTable = table(); % Initalize table
+Power_SettingsTable = table(); % Initalize table
+FFT_SettingsTable = table(); % Initalize table
+
+recordCounter = 1; % Initalize counter for records in DeviceSetting
+entryNumber_TD = 1; % Initialize counter for populating entries to table
+entryNumber_Power = 1; % Initialize counter for populating entries to table
+entryNumber_FFT = 1; % Initialize counter for populating entries to table
+
+inStream_TD = 0; % Initalize as streaming off
+inStream_Power = 0; % Initalize as streaming off
+inStream_FFT = 0; % Initalize as streaming off
+
+changeToInStream = 0;
+streamStartCounter_TD = 1; % Initalize counter for streaming starts
+streamStopCounter_TD = 1; % Initalize counter for streaming stops
+streamStartCounter_Power = 1; % Initalize counter for streaming starts
+streamStopCounter_Power = 1; % Initalize counter for streaming stops
+streamStartCounter_FFT = 1; % Initalize counter for streaming starts
+streamStopCounter_FFT = 1; % Initalize counter for streaming stops
+
+while recordCounter <= length(DeviceSettings)
+    currentSettings = DeviceSettings{recordCounter};
+    
+    % Check if SensingConfig was updated
+    if isfield(currentSettings,'SensingConfig')
+        
+        % If timeDomain updated in this record, create
+        % new row in TD_SettingsTable and populate with metadata
+        if isfield(currentSettings.SensingConfig,'timeDomainChannels')
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            TD_SettingsTable.action{entryNumber_TD} = 'Sense Config';
+            TD_SettingsTable.recNum(entryNumber_TD) = NaN;
+            TD_SettingsTable.time{entryNumber_TD} = HostUnixTime;
+            
+            % Settings will remain in TDsettings until updated
+            TDsettings = convertTDcodes(currentSettings.SensingConfig.timeDomainChannels);
```

Comment:
Adding an additional function call addTableEntry for something generic, or addTDTableEntry might be nice to make this code more readable. This is only a suggestion and is not necessary.

Reply:
[Same comment as below] Do you think it is worth creating a separate function for each TD, power, and FFT to create the toAdd struct? (Which then gets appended to the *_SettingsTable). I didn't do this initially because we'd be passing so much info into the function, but might make it more readable?